### PR TITLE
Update dependencies, kick off [webflux] builds

### PR DESCRIPTION
**Pull request by mraible**

### CODE REVIEW

**Comment by mraible** [comment]
Webflux builds still seem pretty flaky. A common error is:

```
[ERROR] Tests run: 6, Failures: 0, Errors: 6, Skipped: 0, Time elapsed: 62.359 s <<< FAILURE! - in io.github.jhipster.sample.security.DomainUserDetailsServiceIT
[ERROR] io.github.jhipster.sample.security.DomainUserDetailsServiceIT.assertThatEmailIsPrioritizedOverLogin  Time elapsed: 0.044 s  <<< ERROR!
java.lang.IllegalStateException: Failed to load ApplicationContext
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'mongockInitializingBeanRunner' defined in class path resource [io/github/jhipster/sample/config/DatabaseConfiguration.class]: Unsatisfied dependency expressed through method 'mongockInitializingBeanRunner' parameter 1; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.springframework.data.mongodb.core.MongoTemplate' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.springframework.data.mongodb.core.MongoTemplate' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
```

**Comment by mraible** [comment]
@cbornet Any idea why `webflux-mongodb` is failing? It passes fine for me when I run it locally.

```
$ ./mvn clean verify
...
[INFO] Results:
[INFO]
[INFO] Tests run: 310, Failures: 0, Errors: 0, Skipped: 0
```

Maybe we need to enable Testcontainers?

**Comment by DanielFran** [comment]
@mraible Seems related to https://github.com/cloudyrock/mongock/issues/250

**Comment by mraible** [comment]
Thanks, @DanielFran.

**Comment by dieppa** [comment]
> @mraible Seems related to [cloudyrock/mongock#250](https://github.com/cloudyrock/mongock/issues/250)

I am looking at it today ;)

**Comment by pascalgrimaud** [comment]
Following this https://github.com/cloudyrock/mongock/issues/250#issuecomment-695348528 we'll need to update our documentation at https://www.jhipster.tech/using-mongodb/

**Comment by murdos** [comment]
> 
> 
> Following this [cloudyrock/mongock#250 (comment)](https://github.com/cloudyrock/mongock/issues/250#issuecomment-695348528) we'll need to update our documentation at https://www.jhipster.tech/using-mongodb/

Yes ! And I think we should add their logo in our client-side options grid (as we do for liquibase for example)

**Comment by dieppa** [comment]
Hi, I don't think the issue you're having is related to this [mongock's issue](https://github.com/cloudyrock/mongock/issues/250). That issue seems to be due to a wrong configuration when mocking the MongoTemplate. The stack trace @mraible provides suggests that the MongoTemplate is not injected at all. I haven't seen the code but as you are working with webflux(from the branch's title) I think that maybe you are only providing the artifact `spring-boot-starter-data-mongodb-reactive`. Sorry but Mongock doesn't support reactive yet(it will). You can use it with no problem , but you would need to provide `spring-boot-starter-data-mongodb` too.

Once again, I may be wrong, but I think that is the right direction ;)

**Comment by mraible** [comment]
@dieppa I tried adding `spring-boot-starter-data-mongodb` in test scope when we're using reactive. We'll see if that helps the builds pass.

**Comment by dieppa** [comment]
It seems the pipeline failed due to the comment

```xml
Could not compile build file '/home/runner/work/generator-jhipster/generator-jhipster/app/build.gradle'.
> startup failed:
  build file '/home/runner/work/generator-jhipster/generator-jhipster/app/build.gradle': 202: unexpected token: < @ line 202, column 5.
         <!-- Mongock doesn't support reactive yet, so this dependency is necessary. -->
         ^
  
  1 error
```

**Comment by mraible** [comment]
Thanks, @dieppa. Fixed.

File: generators/server/templates/build.gradle.ejs [hunk 1]
```diff
@@ -397,6 +397,11 @@ dependencies {
     <%_ if (['mongodb', 'cassandra', 'couchbase'].includes(databaseType)) { _%>
     implementation "org.springframework.boot:spring-boot-starter-data-<%= databaseType %><% if (reactive) { %>-reactive<% } %>"
     <%_ } _%>
+    <%_ if (databaseType === 'mongodb' && reactive) { _%>
+    // Mongock doesn't support reactive yet, so this dependency is necessary.
+    // See https://github.com/jhipster/generator-jhipster/pull/12536#issuecomment-696683592 for more information.
+    testImplementation "org.springframework.boot:spring-boot-starter-data-mongodb"
```
Comment (mshima) [inline generators/server/templates/build.gradle.ejs:403]:
```suggestion
    implementation "org.springframework.boot:spring-boot-starter-data-mongodb"
```

File: generators/server/templates/pom.xml.ejs [hunk 2]
```diff
@@ -640,6 +640,15 @@
         </dependency>
 <%_ } _%>
 <%_ if (reactive) { _%>
+    <%_ if (databaseType === 'mongodb') { _%>
+        <!-- Mongock doesn't support reactive yet, so this dependency is necessary. -->
+        <!-- See https://github.com/jhipster/generator-jhipster/pull/12536#issuecomment-696683592 for more information. -->
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-mongodb</artifactId>
+            <scope>test</scope>
```
Comment (mshima) [inline generators/server/templates/pom.xml.ejs:649]:
```suggestion
```

File: generators/server/templates/build.gradle.ejs (hunk 1 above)
Comment (mraible) [reply generators/server/templates/build.gradle.ejs:403]: I thought Mongock was only for testing. Will fix.

**Comment by dieppa** [comment]
💯

**Review by mshima — APPROVED** [review]

**Pull request merged**